### ISSUE TITLE
NullReferenceException near Microsoft.Build.Logging.StructuredLogger.MessageProcessor.AddMessage(LazyFormattedBuildEventArgs, String)

### ISSUE THREAD

**Issue by aolszowka** [KirillOsenkov/MSBuildStructuredLog#360]
Attempting to track down an issue in our build system we encounter the following error:

```text
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Build.Logging.StructuredLogger.MessageProcessor.AddMessage(LazyFormattedBuildEventArgs args, String message)
   at Microsoft.Build.Logging.StructuredLogger.Construction.CustomEventRaised(Object sender, CustomBuildEventArgs args)
```

This might be a duplicate of #351 however the issue is marked resolved as of 2.1.133 we are on 2.1.140. We have the log can can provide it privately if needed.

Thank you

**Comment by KirillOsenkov** [comment]
Yes please, send me the log file at kirill dot osenkov at microsoft, or via Twitter DM https://twitter.com/KirillOsenkov

**Comment by aolszowka** [comment]
Email Sent with Link

**Comment by KirillOsenkov** [comment]
@aolszowka this is a .buildlog file. Do you have the original .binlog?

**Comment by aolszowka** [comment]
I do not believe I have the original, I thought they were the same.

I see now that I need to pass `/bl`.

I have done so with the original project and arguments, however opening up the .binlog does not appear to reproduce the issue? Is that expected?

**Comment by KirillOsenkov** [comment]
Yes, the .binlog files are the modern alternative to be used, and they should work fine. I'm updating the viewer to avoid creating the old .buildlog format and switch to using .binlog everywhere.

**Comment by KirillOsenkov** [comment]
I've just published 2.1.147 which uses `/bl` even when you're building in-app. This should hopefully prevent these errors from happening again.
